src/internal/kopia/upload.go

```diff
@@ -993,6 +1012,21 @@ func logBaseInfo(ctx context.Context, m ManifestEntry) {
 		"base_backup_id", mbID)
 }
 
+type mergeStats struct {
```

Comment:
What's the possibility of having this replaced with a [counter](https://github.com/alcionai/corso/blob/main/src/pkg/count/count.go) instead of a new struct?  I'm hoping to centralize stats aggregation around that package in the long term.

Reply:
a few things:

I don't have a preference one way or the other as to making a new `counter` here vs. a new struct

More broadly, I think the question really turns into do these stats need to be reported at the top-level? From my perspective, I see this mostly as a sanity check/debug logging that's good to have all the time and isn't that noisy. If we start talking about passing in a `counter` to the hierarchy merging code so that things can be aggregated at a higher level, I don't find that useful unless we want to start reporting these metrics in some way. The counters do feel a bit more heavy weight than necessary given that they're thread-safe and this code is single threaded, but that's likely me trying to over-optimize things that don't matter since kopia may be making calls to S3 for directory listings anyway

Reply:
> I think the question really turns into do these stats need to be reported at the top-level?

Personally, I don't believe the method of reporting is an important part of the consideration.  Counters are useful for when we want to feed stats up the chain, true.  But more simply it's a way to aggregate behavior without needing to worry about the smaller details.  Such as...

> The counters do feel a bit more heavy weight than necessary given that they're thread-safe and this code is single threaded

The goal is for the interface to be lightweight enough that implementers don't need to worry about heavy or lightweightness.  If we have to go through a decision process of "do I really need the extra utility?" every time we could introduce a counter, then the package is already dead.  What I want to establish is deadpan boilerplate around, "I need to count things, here's a thing that counts things, I call Inc() when I up the value."  Then the implementer is insulated against future changes like moving towards routines, without ever needing to worry about it.  Things should just work.